### PR TITLE
Properly use second argument of fix_permissions

### LINKED ISSUE #1577: Permission error during HABApp installation
## Issue information:
HABApp has no write access to ``etc/openhab`` after a fresh installation of openhabian-pi-raspios32-v1.6.5 on a pi4 4GB.

```
Sep 08 08:59:28 openhabian habapp[3551]:   File "/usr/lib/python3.7/pathlib.py", line 1039, in _opener
Sep 08 08:59:28 openhabian habapp[3551]:     return self._accessor.open(self, flags, mode)
Sep 08 08:59:28 openhabian habapp[3551]: PermissionError: [Errno 13] Permission denied: '/etc/openhab/habapp/logging.yml'
```

Setup has created the HABApp folder with user ``775`` and Group ``openhabian``.
grafik

Applying "fix permission does resolve the issue
grafik

## System information:
```
 openHABian Configuration Tool — [main]{2021-09-07T11:36:31+02:00}(37d24fd)                                       │
                               │ openHAB 3.1.0 - Release Build  
```

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
closes #1577

Signed-off-by: Sebastian Gerber <git@sgerber.de>

### CODE REVIEW

File: functions/habapp.bash [hunk 1]
```diff
@@ -49,8 +49,7 @@ habapp_setup() {
 
   echo -n "$(timestamp) [openHABian] Creating HABApp configuration folder ... "
   if ! cond_redirect mkdir -p "$configFolder"; then echo "FAILED"; return 1; fi
-  if ! cond_redirect chown -R "openhab:${username:-openhabian}" "$configFolder"; then echo "FAILED"; return 1; fi
-  if cond_redirect fix_permissions "$configFolder" 775 775; then echo "OK"; else echo "FAILED"; return 1; fi
+  if cond_redirect fix_permissions "$configFolder" "openhab:${username:-openhabian}" 775 775; then echo "OK"; else echo "FAILED"; return 1; fi
```
Comment (spacemanspiff2007) [inline functions/habapp.bash:52]: Does the 755 have to be "755" ?

**Review by spacemanspiff2007 — COMMENTED** [review]
If I understand fix_permission correctly I can remove the chown call since it sets the permissions recursively.
Is that correct?

File: functions/habapp.bash (hunk 1 above)
Comment (mstormi) [reply functions/habapp.bash:52]: Yes it's recursive. 755 is probably ok, too, and a little bit safer.

Comment (spacemanspiff2007) [reply functions/habapp.bash:52]: Perfect - Then I think this looks good.

**Review by ecdye — DISMISSED** [review]
LGTM

**Comment by ecdye** [comment]
@spacemanspiff2007 Please follow the instructions under the DCO check to fix the issue. Once the DCO is fixed I will merge.

**Comment by spacemanspiff2007** [comment]
@ecdye 
While looking through the history I noticed you (removed the pip part) and it's not clear to me why.
Could you explain it to me?
```
  if ! cond_redirect python3 -m pip install --upgrade pip habapp; then echo "FAILED"; return 1; fi
```
to 
```
  if ! cond_redirect python3 -m pip install --upgrade habapp; then echo "FAILED"; return 1; fi
```

**Comment by ecdye** [comment]
We want apt to be upgrading pip, not python, does it cause an issue with HABapp?

**Comment by spacemanspiff2007** [comment]
> 
> 
> We want apt to be upgrading pip, not python, does it cause an issue with HABapp?

It's only upgrading pip in the newly created venv.
If the user hasn't upgraded in a while HABApp won't install if pip is below a certain version.
This ensures that the installation always possible even if the user was lazy with update.

**Comment by spacemanspiff2007** [comment]
I somehow got the rebase wrong and deleted all changes

**Comment by spacemanspiff2007** [comment]
@ecdye I added the change I remembered but the rest got lost.
Failing test is homegear so imho this is good to be merged

**Comment by spacemanspiff2007** [comment]
I followed the exact steps in the DCO and now it deleted all changes again.

**Comment by spacemanspiff2007** [comment]
@ecdye 
I give up -

**Comment by spacemanspiff2007** [comment]
@ecdye 
I don't know how to get the DCO to pass and the steps suggested are not working.
Is there anywhere a good explanation how to set everything up properly?
Otherwise we'll just leave it buggy and you can close this PR - this has been super frustrating for me especially since we've lost the changes from the first PR.

**Review by ecdye — APPROVED** [review]
DCO appears to be misbehaving, since you have added it to the commit I will override it and merge.

**Pull request merged**